### ISSUE TITLE
Random failures for firecracker k8s tests

### ISSUE THREAD

**Issue by GabyCT** [kata-containers/tests#3922]
I have seen random failures at the firecracker CI with the k8s test at the following line
```
15:00:54 [reset] Unmounting mounted directories in "/var/lib/kubelet"
15:00:55 W0901 20:00:54.999068   17365 unmount_linux.go:48] [reset] Failed to unmount mounted directory in /var/lib/kubelet/: /var/lib/kubelet/plugins/kubernetes.io~local-volume/volumeDevices/block-loop-pv/0228a74f-a07a-4579-a027-953d6045f487
15:00:55 Build timed out (after 15 minutes). Marking the build as aborted.
15:15:55 Build was aborted
```
see http://jenkins.katacontainers.io/job/kata-containers-2.0-tests-firecracker-ubuntu-PR/642/console
http://jenkins.katacontainers.io/job/kata-containers-2.0-firecracker-ubuntu-PR/1038/console
http://jenkins.katacontainers.io/job/kata-containers-2.0-firecracker-ubuntu-PR/1036/
/cc @fidencio @jrieck1991

**Comment by GabyCT** [comment]
From the logs I can see that it is having the issues at https://github.com/kata-containers/tests/blob/6d0499b1b1a6d668967a2a0b8c79a2fb86c1c514/integration/kubernetes/k8s-block-volume.bats because it is saying 
```
22:17:57 W0831 03:17:57.684259   17456 unmount_linux.go:48] [reset] Failed to unmount mounted directory in /var/lib/kubelet/: /var/lib/kubelet/plugins/kubernetes.io~local-volume/volumeDevices/block-loop-pv/b7f9ed23-35d2-4c63-ad5f-05f414d6bb34
```

**Comment by c3d** [comment]
@GabyCT During triage, I asked why this would show up only for firecracker. @jcvenegas pointed out that this may be because FC uses a special way to mount block devices, does not use virtiofs.

**Comment by bergwolf** [comment]
How is this resolved? I'm getting it when testing stable-2.2 but looks like it is fixed on main?
